### PR TITLE
Documentation: Improve cilium-cli and hubble cli installation instructions

### PR DESCRIPTION
**Pull request by chancez**
- Improve the CLI install by failing fast if the binary is missing (can happen if stable.txt is bumped before 
the new release comes out)
- Download correct binary on Linux based on the architecture.

```release-note
Documentation: Improve cilium-cli and hubble cli installation instructions
```

### CODE REVIEW

**Comment by chancez** [comment]
I think this could be nice to get into the 1.12 release as well.

**Comment by chancez** [comment]
Noticed similar patterns with hubble CLI so I updated those also.

File: Documentation/gettingstarted/cli-download.rst [hunk 1]
```diff
@@ -8,30 +8,25 @@ various features (e.g. clustermesh, Hubble).
     .. code-block:: shell-session
 
       CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/master/stable.txt)
-      curl -L --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-linux-amd64.tar.gz{,.sha256sum}
-      sha256sum --check cilium-linux-amd64.tar.gz.sha256sum
-      sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/local/bin
-      rm cilium-linux-amd64.tar.gz{,.sha256sum}
+      CLI_ARCH=$(arch)
```
Comment (tklauser) [inline Documentation/gettingstarted/cli-download.rst]: It seems not all distributions include the `arch` command in their coreutils package. For example I don't have it on Arch Linux (oh the irony). Could we use `uname -m` instead for Linux?

File: Documentation/gettingstarted/cli-download.rst [hunk 2]
```diff
@@ -8,30 +8,25 @@ various features (e.g. clustermesh, Hubble).
...
-  .. group-tab:: macOS (Apple Silicon - ARM)
-
-    .. code-block:: shell-session
-
-      CILIUM_CLI_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/cilium-cli/master/stable.txt)
-      curl -L --remote-name-all https://github.com/cilium/cilium-cli/releases/download/${CILIUM_CLI_VERSION}/cilium-darwin-arm64.tar.gz{,.sha256sum}
-      shasum -a 256 -c cilium-darwin-arm64.tar.gz.sha256sum
-      sudo tar xzvfC cilium-darwin-arm64.tar.gz /usr/local/bin
-      rm cilium-darwin-arm64.tar.gz{,.sha256sum}
+      CLI_ARCH=$(arch)
```
Comment (tklauser) [inline Documentation/gettingstarted/cli-download.rst]: Will this determine the correct arch when the shell and/or terminal is an universal binary? See https://github.com/cilium/cilium/pull/19009#pullrequestreview-898661284 for previous discussion.

File: Documentation/gettingstarted/cli-download.rst (hunk 1 above)
Comment (chancez) [reply Documentation/gettingstarted/cli-download.rst]: Yeah Robin mentioned this as well. I will try using `uname` for everything

File: Documentation/gettingstarted/cli-download.rst (hunk 2 above)
Comment (chancez) [reply Documentation/gettingstarted/cli-download.rst]: Yeah, `arch` returns i386 when running under Rosetta, but `uname -m` returns arm64 as expected. This makes sense to me.

**Comment by chancez** [comment]
@tklauser Ok I've updated it to use `uname -m` and simplified the case statements slightly as well. If someone could test on amd64 Linux and amd64 MacOS that would be ideal.

**Review by tklauser — APPROVED** [review]
LGTM, tested on linux/amd64, linux/arm64 and darwin/amd64.

File: Documentation/gettingstarted/hubble-install.rst [hunk 3]
```diff
@@ -19,10 +21,12 @@
       .. code-block:: shell-session
 
          export HUBBLE_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/hubble/master/stable.txt)
-         curl -L --fail --remote-name-all https://github.com/cilium/hubble/releases/download/$HUBBLE_VERSION/hubble-darwin-amd64.tar.gz{,.sha256sum}
-         shasum -a 256 -c hubble-darwin-amd64.tar.gz.sha256sum
-         sudo tar xzvfC hubble-darwin-amd64.tar.gz /usr/local/bin
-         rm hubble-darwin-amd64.tar.gz{,.sha256sum}
+         HUBBLE_ARCH=amd64
+         if [ "${HUBBLE_ARCH}" = "amd64" ]; then HUBBLE_ARCH=arm64; fi
```
Comment (qmonnet) [inline Documentation/gettingstarted/hubble-install.rst]: `if [ "${HUBBLE_ARCH}"` should be `if [ "$(uname -m)"`?

And `amd64` should be `arm64`? Or did I miss something?

File: Documentation/gettingstarted/hubble-install.rst [hunk 4]
```diff
@@ -7,10 +7,12 @@
       .. code-block:: shell-session
 
          export HUBBLE_VERSION=$(curl -s https://raw.githubusercontent.com/cilium/hubble/master/stable.txt)
-         curl -L --fail --remote-name-all https://github.com/cilium/hubble/releases/download/$HUBBLE_VERSION/hubble-linux-amd64.tar.gz{,.sha256sum}
-         sha256sum --check hubble-linux-amd64.tar.gz.sha256sum
-         sudo tar xzvfC hubble-linux-amd64.tar.gz /usr/local/bin
-         rm hubble-linux-amd64.tar.gz{,.sha256sum}
+         HUBBLE_ARCH=amd64
+         if [ "${HUBBLE_ARCH}" = "aarch64" ]; then HUBBLE_ARCH=arm64; fi
```
Comment (qmonnet) [inline Documentation/gettingstarted/hubble-install.rst]: `if [ "${HUBBLE_ARCH}"` should be `if [ "$(uname -m)"`?

**Review by qmonnet — CHANGES_REQUESTED** [review]
Looks good, save for what I think is a mistake on Hubbe install arch conditions. Apologies for taking so long to review

File: Documentation/gettingstarted/hubble-install.rst (hunk 4 above)
Comment (chancez) [reply Documentation/gettingstarted/hubble-install.rst]: Wow, yeah I wonder how this worked for me when I tested. Maybe cause my Mac can run both binaries. Didn't pay enough attention. Ill fix this.

**Comment by chancez** [comment]
Travis tests hit a panic on some BGP related code it seems.

**Comment by qmonnet** [comment]
> Travis tests hit a panic on some BGP related code it seems.

The fix has been merged on `master`: https://github.com/cilium/cilium/pull/20521

**Review by qmonnet — APPROVED** [review]
Looks better, thank you!

**Pull request merged**